startlxqtwayland.in

```diff
@@ -65,9 +65,29 @@ export QT_ACCESSIBILITY=1
 # use lxqt-applications.menu for main app menu
 export XDG_MENU_PREFIX="lxqt-"
 
+share_dir="$(dirname $(dirname "$0"))"/share
+
 if [ ! -d "$XDG_CONFIG_HOME/lxqt/wayland" ]; then
     mkdir -p $XDG_CONFIG_HOME/lxqt/wayland/
 fi
+# Check for default compositor and screenlocker eventually set by distribution
+if grep -q "compositor" "$share_dir/lxqt/session.conf"; then
+    default_compositor=$(cat $share_dir/lxqt/session.conf|grep compositor)
+    default_screenlocker=$(cat $share_dir/lxqt/session.conf|grep lock_command_wayland)
+    if [ ! -f "$XDG_CONFIG_HOME/lxqt/session.conf" ]; then
+        cp -a $share_dir/lxqt/ $XDG_CONFIG_HOME/
+    else
+        config_file="$XDG_CONFIG_HOME/lxqt/session.conf"
+        if ! grep -q "compositor" "$XDG_CONFIG_HOME/lxqt/session.conf"; then
```

Comment:
Isn't it better to search for "compositor=" instead of "compositor" (and the same for "lock_command_wayland" below)?

Reply:
It would be the same, no? What could be prevented is taking `compositor=  ` as configured, see https://github.com/lxqt/lxqt-wayland-session/pull/59#issuecomment-2644715321

Reply:
OK.